src/AbstractArray.php

```diff
@@ -180,7 +180,7 @@ public function __construct(array $elements = [])
      *
      * @return static The instance array with combined elements
      */
-    abstract public function combineTo(array $array);
+    abstract public function useAndCombineTo(array $array);
```

Comment:
Okay so if I follow correctly what you are saying then 

```php
//replaceWith does this
$res1 = $current->replaceWith($new)
//and useAndReplaceIn  does
$res2 = (MutableArray($new))->replaceWith($current);
//so that
var_dump($res1 == $res2) //should return true
```
If this is the case then IMHO you should not even create `useAndReplaceIn` in the first place because it unnecessary confuse people while implementing it in user land is straight forward :+1: